### ISSUE TITLE
lint: ignored statements in a bench-loop

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#542]
Name: `falseBench`
Before:
```go
func BenchFoo(b *testing.B) {
    for i := 0; i < b.N; i++ {
        foo()
    }
}
```
After:
```go
func BenchFoo(b *testing.B) {
    for i := 0; i < b.N; i++ {
        _ = foo()
    }
}
```

**Comment by Quasilyte** [comment]
Actually, `_ = foo()` not always help.
Assigning to global var does though.

This topic also was highlighted on the gophercon UK. :)

**Comment by Quasilyte** [comment]
Probably should be called `noopBench` (or `nopBench`, see [NOP](https://ru.wikipedia.org/wiki/NOP)).

**Comment by cristaloleg** [comment]
Updated, thanks.
